### PR TITLE
User sermon note backend

### PR DESCRIPTION
**Pull request by Baeck016**

### CODE REVIEW

**Review by alvone01 — APPROVED** [review]
LGTM contingent on Midnite

**Comment by Midnite1212** [comment]
> LGTM contingent on Midnite
你不好

File: server/api/controllers/userSermonNotes/update-user-sermon-notes.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    { sermonId, editedContent, themes, stickyNote, isDeleted },
+    exits
+  ) {
+    const user = this.req.user.fullName;
+    sails.log.info(`${user}: Updating user sermon note: ${sermonId}`);
+
+    if (sermonId) {
+      try {
+        let data = await UserSermonNotes.updateOne({
+          sermonId: sermonId,
```
Comment (Midnite1212) [inline server/api/controllers/userSermonNotes/update-user-sermon-notes.js:48]: Don't do this, this is searching for all the sermon note with the same sermon, not the user's specific sermon note. You should add another params in the function to accept id, and pass it here to seach for the specific user's sermon note id in mongo
`_id:userSermonNoteId`

or better yet, you can do what you did in the get api, use the sermonId and userId to find the specific sermon note you want. Rmb updateOne only returns one object, not an array of objects like find().

File: server/api/controllers/userSermonNotes/get-user-sermon-notes.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    },
+  },
+  fn: async function ({ sermonId, userId }, exits) {
+    try {
+      if (sermonId && userId) {
+        let data = await UserSermonNotes.find({
+          sermonId: sermonId,
+          userId: userId,
+          isDeleted: false,
+        });
```
Comment (Midnite1212) [inline server/api/controllers/userSermonNotes/get-user-sermon-notes.js:33]: searching like this is good, should be easier for fe ppl to implement too 👍

File: server/api/controllers/userSermonNotes/delete-user-sermon-notes.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      description: 'Please fill in the required fields.',
+    },
+  },
+
+  fn: async function ({ sermonId }, exits) {
+    if (sermonId) {
+      try {
+        let data = await UserSermonNotes.updateOne({
+          sermonId: sermonId,
+          isDeleted: false,
```
Comment (Midnite1212) [inline server/api/controllers/userSermonNotes/delete-user-sermon-notes.js:32]: same comment with update

**Review by Midnite1212 — CHANGES_REQUESTED** [review]
faiti

File: server/api/models/UserSermonNotes.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      model: 'User',
+      required: true,
+    },
+    editedContent: {
+      type: 'String',
+      allowNull: true,
+    },
+    themes: {
+      type: 'String',
+      isIn: ['Forgiven', 'Redemption'],
```
[the remark this answers is not in the document]
Comment (Baeck016) [reply server/api/models/UserSermonNotes.js:18]: owhh forgot to change this. This is just a dummy, because wanted to test it out

**Review by Midnite1212 — APPROVED** [review]

**Pull request merged**